### PR TITLE
Uml 1583:  Admin portal logout

### PR DESCRIPTION
**Pull request by SeemaMenon**
# Purpose

Fixes UML-1583

Introduced new handers to redirect users to cognito log out URL and then to the log out page.

_Explain how your code addresses the purpose of the change_

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Use a Lasting Power of Attorney service_

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] New event_codes have been documented on the [wiki page](https://opgtransform.atlassian.net/wiki/spaces/LSML2/pages/3277881441/Understanding+the+event+logs)
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] I have added welsh translation tags and updated translation files
* [ ] I have run an accessibility tool on any pages I have made changes to and fixed any issues found
* [ ] I have notified the Interaction Designer of any content changes so that appropriate screenshots/flow diagram changes can be made
* [ ] The product team have tested these changes

### CODE REVIEW

File: terraform/environment/admin_ecs.tf [hunk 1]
```diff
@@ -210,6 +210,10 @@ locals {
         {
           name  = "ADMIN_JWT_SIGNING_KEY_URL",
           value = "https://public-keys.auth.elb.eu-west-1.amazonaws.com"
+        },
+        {
+          name  = "ADMIN_CLIENT_ID",
+          value = "${aws_cognito_user_pool_client.use_a_lasting_power_of_attorney_admin.id}"
```
Comment (andrewpearce-digital) [inline terraform/environment/admin_ecs.tf:216]:
```suggestion
          value = "${aws_cognito_user_pool_client.use_a_lasting_power_of_attorney_admin.id[0]}"
```

File: terraform/environment/admin_ecs.tf [hunk 2]
```diff
@@ -210,6 +210,10 @@ locals {
         {
           name  = "ADMIN_JWT_SIGNING_KEY_URL",
           value = "https://public-keys.auth.elb.eu-west-1.amazonaws.com"
+        },
+        {
+          name  = "ADMIN_CLIENT_ID",
+          value = "${aws_cognito_user_pool_client.use_a_lasting_power_of_attorney_admin.id[0]}"
```
Comment (andrewpearce-digital) [inline terraform/environment/admin_ecs.tf:216]:
```suggestion
          value = "${aws_cognito_user_pool_client.use_a_lasting_power_of_attorney_admin[0].id}"
```

File: service-admin/internal/server/handlers/logout.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+			Scheme: "https",
+			Host: r.Host,
+			Path: "/",
+		}
+		
+		v := cognitoLogoutURL.Query()
+		v.Set("logout_uri", l.String())
+		cognitoLogoutURL.RawQuery = v.Encode()
+	
+		http.Redirect(w, r, cognitoLogoutURL.String(), 301)
```
Comment (cooperaj) [inline service-admin/internal/server/handlers/logout.go:35]: Fix as suggested by the linter. :)

**Review by cooperaj — CHANGES_REQUESTED** [review]

**Review by andrewpearce-digital — CHANGES_REQUESTED** [review]
one change needed in service-admin/internal/server/handlers/logout.go as advised by the linter

**Review by andrewpearce-digital — APPROVED** [review]

**Pull request merged**